src/Processors/QueryPlan/Optimizations/optimizeUseAggregateProjection.cpp

```diff
@@ -589,6 +589,7 @@ bool optimizeUseAggregateProjections(QueryPlan::Node & node, QueryPlan::Nodes &
     }
     else if (!candidates.real.empty())
     {
+        reading->applyFilters();
```

Comment:
Let's understand why this is needed.
Ideally, maybe we can reorder some plan optimizations to avoid it